src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -4178,18 +4178,32 @@ class CreateSequenceOptionsSegment(ansi.CreateSequenceOptionsSegment):
     match_grammar = OneOf(
         Sequence("AS", Ref("DatatypeSegment")),
         Sequence(
-            "INCREMENT", Ref.keyword("BY", optional=True), Ref("NumericLiteralSegment")
+            "INCREMENT",
+            Ref.keyword("BY", optional=True),
+            Ref("NegativeSegment", optional=True),
```

Comment:
The `NegativeSegment` should be replaced with `SignedSegmentGrammar` as values like `+4` are also allowed.